astropy/coordinates/builtin_frames/fk4.py

```diff
@@ -169,42 +169,34 @@ def fk4_e_terms(equinox):
 @frame_transform_graph.transform(FunctionTransform, FK4, FK4NoETerms)
 def fk4_to_fk4_no_e(fk4coord, fk4noeframe):
     # Extract cartesian vector
-    c = fk4coord.cartesian.xyz
-    r = np.asarray(c.reshape((3, c.size // 3)))
+    r = fk4coord.cartesian
 
     # Find distance (for re-normalization)
-    d_orig = np.sqrt(np.sum(r ** 2, axis=0))
+    d_orig = abs(r)
     r /= d_orig
 
     # Apply E-terms of aberration. Note that this depends on the equinox (not
     # the observing time/epoch) of the coordinates. See issue #1496 for a
     # discussion of this.
-    eterms_a = np.asarray(fk4_e_terms(fk4coord.equinox))
-    r = r - eterms_a.reshape(3, 1) + np.dot(eterms_a, r) * r
+    eterms_a = CartesianRepresentation(
+        u.Quantity(fk4_e_terms(fk4coord.equinox), u.dimensionless_unscaled,
+                   copy=False), copy=False)
+    r = r - eterms_a + eterms_a.dot(r) * r
 
     # Find new distance (for re-normalization)
-    d_new = np.sqrt(np.sum(r ** 2, axis=0))
+    d_new = abs(r)
 
     # Renormalize
     r *= d_orig / d_new
 
-    subshape = c.shape[1:]
-    x = r[0].reshape(subshape)
-    y = r[1].reshape(subshape)
-    z = r[2].reshape(subshape)
-
     #now re-cast into an appropriate Representation, and precess if need be
     if isinstance(fk4coord.data, UnitSphericalRepresentation):
-        representation = CartesianRepresentation(x=x*u.one, y=y*u.one, z=z*u.one)
-        representation = representation.represent_as(UnitSphericalRepresentation)
-    else:
-        representation = CartesianRepresentation(x=x*c.unit, y=y*c.unit, z=z*c.unit)
+        r = r.represent_as(UnitSphericalRepresentation)
```

Comment:
I realize you're following `r` from some of the other cases here, but what about uniformly changing to `representation` instead of `r`?  I remember when writing some of this that I avoided `r` because it seemed like r as in the spherical coordinate r rather than "representation".